test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/BodyModelBinderTests.cs

```diff
@@ -206,6 +205,88 @@ public class BodyModelBinderTests
             Assert.Same(canReadFormatter1, bindingContext.Result.Model);
         }
 
+        [Fact]
+        public async Task BindModelAsync_LogsFormatterRejectionAndSelection()
+        {
+            // Arrange
+            var sink = new TestSink();
+            var loggerFactory = new TestLoggerFactory(sink, enabled: true);
+            var inputFormatters = new List<IInputFormatter>()
+            {
+                new TestInputFormatter(canRead: false),
+                new TestInputFormatter(canRead: true),
+            };
+
+            var provider = new TestModelMetadataProvider();
+            provider.ForType<Person>().BindingDetails(d => d.BindingSource = BindingSource.Body);
+            var bindingContext = GetBindingContext(typeof(Person), metadataProvider: provider);
+            bindingContext.HttpContext.Request.ContentType = "application/json";
+            var binder = new BodyModelBinder(inputFormatters, new TestHttpRequestStreamReaderFactory(), loggerFactory);
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.Equal("Rejected input formatter 'Microsoft.AspNetCore.Mvc.ModelBinding.Binders.BodyModelBinderTests+TestInputFormatter' for content type 'application/json'", sink.Writes[0].State.ToString());
```

Comment:
@jbagga just use string interopolation if you want to shorten this. My earlier complaint was not about string interpolation in general, just when it includes no expression.